From d130ab0eeb33b6acbb40173665f4a6ca1e07ec19 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 22 Jul 2020 09:58:42 -0700
Subject: [PATCH 1452/1851] drm: xlnx: zynqmp_dp: include zynqmp_dp.h and fix
 the return type

commit 06cf477a2c9dbaf5ca33769ba311ce6f9340e8d0 from
https://github.com/Xilinx/linux-xlnx.git

Otherwise there are warnings as below:

drivers/gpu/drm/xlnx/zynqmp_dp.c:411:5: warning: no previous prototype for 'zynqmp_dp_set_color' [-Wmissing-prototypes]
  411 | int zynqmp_dp_set_color(struct zynqmp_dp *dp, const char *color)
      |     ^~~~~~~~~~~~~~~~~~~
...

The return type of zynqmp_dp_set_color() was incorrect and fixed.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Dylan Yip <dylan.yip@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_dp.c | 1 +
 drivers/gpu/drm/xlnx/zynqmp_dp.h | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_dp.c b/drivers/gpu/drm/xlnx/zynqmp_dp.c
index 7dfee9993a94..eebf7e6a9d0e 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dp.c
@@ -36,6 +36,7 @@
 #include <linux/uaccess.h>
 
 #include "zynqmp_disp.h"
+#include "zynqmp_dp.h"
 #include "zynqmp_dpsub.h"
 
 static uint zynqmp_dp_aux_timeout_ms = 50;
diff --git a/drivers/gpu/drm/xlnx/zynqmp_dp.h b/drivers/gpu/drm/xlnx/zynqmp_dp.h
index 2f6ce3f3e8cf..33cf4532d935 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dp.h
+++ b/drivers/gpu/drm/xlnx/zynqmp_dp.h
@@ -22,7 +22,7 @@
 struct zynqmp_dp;
 struct drm_display_mode;
 
-const int zynqmp_dp_set_color(struct zynqmp_dp *dp, const char *color);
+int zynqmp_dp_set_color(struct zynqmp_dp *dp, const char *color);
 void zynqmp_dp_enable_vblank(struct zynqmp_dp *dp);
 void zynqmp_dp_disable_vblank(struct zynqmp_dp *dp);
 void zynqmp_dp_encoder_mode_set_stream(struct zynqmp_dp *dp,
-- 
2.31.1

